### PR TITLE
SBERDOMA-495 Added /contact page

### PR DESCRIPTION
**Pull request by SavelevMatthew**
Added contacts section (only overview page ("/contacts") for this PL)
Added helper utils covered with tests
Added table view for contacts (for now 1 Contact = 1 Line)

This is duplicated pull request. It was created to separate big task and decrease amount of code to be reviewed. Previous version: https://github.com/open-condo-software/condo/pull/224

### CODE REVIEW

File: apps/condo/domains/contact/utils/helpers.spec.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    getPageIndexFromQuery,
+    CONTACT_PAGE_SIZE,
+} from './helpers'
+
+describe('Contact Helpers', () => {
+    describe('query utils', () => {
+        describe('searchToQuery', () => {
+            it('correct work on correct string', () => {
+                const testTimes = 1000
+                for (let i = 0; i < testTimes; i++) {
```
Comment (pahaz) [inline apps/condo/domains/contact/utils/helpers.spec.ts:16]: why do you use the `for` loop inside tests? it's a bad practice

File: apps/condo/domains/contact/utils/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+export const searchToQuery = (search?: string): ContactWhereInput[] => {
+    if (!search) {
+        return
+    }
+    return [
+        { name_contains_i: search },
+        { phone_contains_i: search },
+        { email_contains_i: search },
+        { property: { address_contains_i: search } },
+    ].filter(Boolean)
```
Comment (pahaz) [inline apps/condo/domains/contact/utils/helpers.ts:24]: why you need a filter?

File: apps/condo/pages/contact/index.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+import { useRouter } from 'next/router'
+import qs from 'qs'
+import { pickBy, get, debounce } from 'lodash'
+import React, { useCallback } from 'react'
+import { EmptyListView } from '@condo/domains/common/components/EmptyListView'
+import { useTableColumns } from '@condo/domains/contact/hooks/useTableColumns'
+import { useSearch } from '@condo/domains/common/hooks/useSearch'
+import { useOrganization } from '@core/next/organization'
+import { Contact } from '@condo/domains/contact/utils/clientSchema'
+import { Button } from '../../domains/common/components/Button'
```
Comment (pahaz) [inline apps/condo/pages/contact/index.tsx:26]: @condo
```suggestion
import { Button } from '@condo/domains/common/components/Button'
```

**Review by pahaz — APPROVED** [review]

**Comment by AntonAL** [comment]
Could you please name sequential commits differently ;)

File: apps/condo/domains/common/components/Icons.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    return (
+        <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path fillRule="evenodd" clipRule="evenodd" d="M12 13C14.7614 13 17 10.7614 17 8C17 5.23858 14.7614 3 12 3C9.23857 3 6.99999 5.23858 6.99999 8C6.99999 10.7614 9.23857 13 12 13ZM12 13C18.6328 13 20.4538 18.843 20.8824 20.8401C20.9578 21.1913 20.6844 21.5 20.3253 21.5H3.67472C3.31556 21.5 3.04218 21.1913 3.11755 20.8401C3.54621 18.8431 5.36714 13 12 13Z" fill="currentColor"/>
+        </svg>
+    )
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const UserIcon = props => {
+    console.log(props)
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/common/components/Icons.tsx]: done 👍

File: apps/condo/domains/common/components/Icons.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import Icon from '@ant-design/icons'
+
+
+const UserIconSVG: React.FC = () => {
+    return (
+        <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path fillRule="evenodd" clipRule="evenodd" d="M12 13C14.7614 13 17 10.7614 17 8C17 5.23858 14.7614 3 12 3C9.23857 3 6.99999 5.23858 6.99999 8C6.99999 10.7614 9.23857 13 12 13ZM12 13C18.6328 13 20.4538 18.843 20.8824 20.8401C20.9578 21.1913 20.6844 21.5 20.3253 21.5H3.67472C3.31556 21.5 3.04218 21.1913 3.11755 20.8401C3.54621 18.8431 5.36714 13 12 13Z" fill="currentColor"/>
+        </svg>
+    )
+}
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/common/components/Icons.tsx]: done 👍

File: apps/condo/domains/contact/utils/helpers.spec.ts [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+    createSorterMap,
+    queryToSorter,
+    getPageIndexFromQuery,
+    CONTACT_PAGE_SIZE,
+} from './helpers'
+
+describe('Contact Helpers', () => {
+    describe('query utils', () => {
+        describe('searchToQuery', () => {
+            it('correct work on correct string', () => {
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/contact/utils/helpers.spec.ts]: done 👍

File: apps/condo/domains/contact/utils/helpers.spec.ts [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+                                            filters[k].result,
+                                        ],
+                                    })
+                                })
+                            }
+                        }
+                    }
+                })
+                describe('if 4 filters defined', () => {
+                    for (let i = 0; i < filters.length - 3; i++) {
```
Comment (dkoviazin) [inline apps/condo/domains/contact/utils/helpers.spec.ts:108]: Looks strange

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/contact/utils/helpers.spec.ts (hunk 7 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/utils/helpers.spec.ts:108]: This set of tests determines possibly of applying all filters in any combinations

**Comment by SavelevMatthew** [comment]
Git required to resolve conflicts, that's the reason of last commit.

File: apps/condo/pages/contact/index.tsx [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+import { pickBy, get, debounce } from 'lodash'
+import React, { useCallback } from 'react'
+import { EmptyListView } from '@condo/domains/common/components/EmptyListView'
+import { useTableColumns } from '@condo/domains/contact/hooks/useTableColumns'
+import { useSearch } from '@condo/domains/common/hooks/useSearch'
+import { useOrganization } from '@core/next/organization'
+import { Contact } from '@condo/domains/contact/utils/clientSchema'
+import { Button } from '@condo/domains/common/components/Button'
+import { SortContactsBy } from '../../schema'
+const ADD_CONTACT_ROUTE = '/contact/create/'
```
Comment (pahaz) [inline apps/condo/pages/contact/index.tsx:28]:
```suggestion

const ADD_CONTACT_ROUTE = '/contact/create/'
```

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/common/components/icons/Icons.tsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import Icon from '@ant-design/icons'
+
+
+const UserIconSVG: React.FC = () => {
```
Comment (pahaz) [inline apps/condo/domains/common/components/icons/Icons.tsx:5]: probably it's better to move it inside `user` domain

Comment (pahaz) [reply apps/condo/domains/common/components/icons/Icons.tsx:5]: And use UserIcon.jsx filename

Comment (SavelevMatthew) [reply apps/condo/domains/common/components/icons/Icons.tsx:5]: @AntonAL suggested to group all custom icons here, so idk...

Comment (pahaz) [reply apps/condo/domains/common/components/icons/Icons.tsx:5]: ok, could anyone add it as rule in our code style?

Comment (SavelevMatthew) [reply apps/condo/domains/common/components/icons/Icons.tsx]: > And use UserIcon.jsx filename

done 👍

File: apps/condo/pages/contact/index.tsx (hunk 8 above)
Comment (SavelevMatthew) [reply apps/condo/pages/contact/index.tsx:29]: Fixed 👍

**Pull request merged**